src/context.js

```diff
@@ -299,6 +307,48 @@ class Context {
 			this.broker.emit("metrics.trace.span.finish", payload);
 		}
 	}
+
+	/**
+	 * Assign extra metrics taking into account action definitions
+	 *
+	 * @param {string} name Field of the context to be assigned.
+	 * @param {any} payload Object for assignement.
+	 *
+	 * @private
+	 * @memberof Context
+	 */
+	_assignExtraMetrics(name, payload) {
+		let def = this.action.metrics[name];
+		// if metrics definitions is boolean do default, metrics=true
+		let toAssign = {};
+		if (_.isBoolean(def) && def) {
+			toAssign[name] = this[name];
+			_.assign(payload, toAssign);
+		} else if (_.isArray(def)) {
+			toAssign[name] = _.pick(this[name], def);
+			_.assign(payload, toAssign);
+		} else if (_.isFunction(def)) {
+			toAssign[name] = def(this[name]);
+			_.assign(payload, toAssign);
```

Comment:
The context metrics is a critical part because it is called for every request/call. So it should be high performance.
I measured and the performance is decreased by -33%.
Master:
```
Suite: Call with statistics & metrics
√ No statistics*                    1,388,993 rps
√ With metrics*                       523,056 rps
√ With metrics & statistics*          475,057 rps
```

This PR:
```
Suite: Call with statistics & metrics
√ No statistics*                    1,492,456 rps
√ With metrics*                       359,542 rps
√ With metrics & statistics*          337,990 rps
```

I think due to `_.assign`. The `lodash` functions are not too good for performance. Could you change it to `Object.assign`?

Reply:
Ok I’ll change it